File: backend/src/main/java/com/alimberdi/backend/portfolio/model/entity/Education.java
package com.alimberdi.backend.portfolio.model.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.UUID;

@Entity
@Table(name = "education", schema = "portfolio")
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Education {

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private UUID id;

	@Column(nullable = false)
	private String institution;

	@Column(nullable = false, columnDefinition = "TEXT")
	private String description;

	@Column(nullable = false)
	private String location;

	@Column(nullable = false)
	private String degree;

	@Column(nullable = false)
	private String speciality;

	@Column(name = "start_date", nullable = false)
	private YearMonth startDate;

	@Column(name = "end_date")
	private YearMonth endDate;

	private BigDecimal grade;

	private String url;

	@Column(name = "order_index", unique = true, nullable = false)
	private Integer orderIndex;

	@CreatedDate
	@Column(nullable = false, updatable = false)
	private Instant createdAt;

}

File: frontend/src/type/portfolio/education.ts
export type EducationStatus = "IN_PROGRESS" | "COMPLETED";

export type YearMonth = {
    year: number,
    month: number
}

export type EducationCreateResponse = {
    institution: string;
    description: string;
    location: string;
    degree: string;
    speciality: string;
    startDate: YearMonth;
    endDate: YearMonth;
    grade: number;
    status: EducationStatus;
    url: string;
}

export type EducationResponse = {
    id: string;
    institution: string;
    description?: string;
    location: string;
    degree: string;
    speciality: string;
    startDate: string;
    endDate?: string;
    grade?: number;
    status: EducationStatus;
    url?: string;
    orderIndex: number;
    createdAt: Date;
}